### PR TITLE
HAWQ-38. Support HA for libyarn

### PR DESCRIPTION
**Pull request by linwen**
Support HA for libyarn. Please review.

### CODE REVIEW

File: depends/libyarn/src/libyarnclient/ApplicationClient.cpp [hunk 1]
```diff
@@ -4,29 +4,146 @@
...
+    	maxRMHARetry = sessionConfig.getRpcMaxHaRetry();
+    }
+
+    if (!enableRMHA)
+    {
+    	appClientProtos.push_back(
+    		std::shared_ptr<ApplicationClientProtocol>(
+    			new ApplicationClientProtocol(user, host, port, tokenService, sessionConfig)));
+    }
+    else {
```
Comment (wangzw) [inline depends/libyarn/src/libyarnclient/ApplicationClient.cpp]: “else" should not start a new line.

File: depends/libyarn/src/libyarnclient/ApplicationClient.cpp [hunk 2]
```diff
@@ -4,29 +4,146 @@
...
+    	/*
+    	 * iterate RMInfo vector and create 1-1 applicationClientProtocol for each standby RM
+    	 */
+		for (size_t i = 0; i < rmInfos.size(); ++i) {
+			appClientProtos.push_back(
```
Comment (wangzw) [inline depends/libyarn/src/libyarnclient/ApplicationClient.cpp]: should not mix tab and space

**Comment by wangzw** [comment]
Code will be OK after fixing some style issues. 
1) should not mix tab and space.
2) "else" should not start a new line.

**Comment by jiny2** [comment]
+1
I reviewed the code changes, They are ok to me.

**Comment by linwen** [comment]
All comments are accepted. Thanks!

**Pull request closed without merging**